features/nfc/nfc/common/URI.h

```diff
@@ -0,0 +1,189 @@
+/* mbed Microcontroller Library
+ * Copyright (c) 2018 ARM Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef NFC_COMMON_URI_H_
+#define NFC_COMMON_URI_H_
+
+#include <nfc/ndef/RecordParser.h>
+#include <nfc/ndef/MessageBuilder.h>
+#include <stdint.h>
+#include <string.h>
+
+#include <platform/Span.h>
+
+namespace mbed {
+namespace nfc {
+namespace common {
+
+/**
+ * @addtogroup nfc
+ * @{
+ */
+
+/**
+ * Model the well known type URI.
+ */
+class URI {
+public:
+    /**
+     * Identifier codes
+     */
+    enum uri_identifier_code_t {
+        NA = 0x00, /// Not applicable
+        HTTP_WWW = 0x01, /// http://www.
+        HTTPS_WWW = 0x02, /// https://www.
+        HTTP = 0x03, /// http://
+        HTTPS = 0x04, /// https://
+        TEL = 0x05, /// tel:
+        MAILTO = 0x06, /// mailto:
+        FTP_ANONYMOUS = 0x07, /// ftp://anonymous:anonymous@
+        FTP_FTP = 0x08, /// ftp://ftp.
+        FTPS = 0x09, /// ftps://
+        SFTP = 0x0A, /// sftp://
+        SMB = 0x0B, /// smb://
+        NFS = 0x0C, /// nfs://
+        FTP = 0x0D, /// ftp://
+        DAV = 0x0E, /// dav://
+        NEWS = 0x0F, /// news:
+        TELNET = 0x10, /// telnet://
+        IMAP = 0x11, /// imap:
+        RSTP = 0x12, /// rstp://
+        URN = 0x13, /// urn:
+        POP = 0x14, /// pop:
+        SIP = 0x15, /// sip:
+        SIPS = 0x16, /// sips:
+        TFTP = 0x17, /// tftp:
+        BTSPP = 0x18, /// btspp://
+        BTL2CAP = 0x19, /// btl2cap://
+        BTGOEP = 0x1A, /// btgoep://
+        TCPOBEX = 0x1B, /// tcpobex://
+        IRDAOBEX = 0x1C, /// irdaobex://
+        FILE = 0x1D, /// file://
```

Comment:
This will clash if `stdio.h` is included

Reply:
It shouldn't, it is not the same scope: FILE is a typedef in the global scope and this is an enum inside the URI class; it is only accessible with URI::FILE.

Reply:
Good point :)